reverse_strand.py

```diff
@@ -0,0 +1,39 @@
+#! /usr/bin/env python3
+import argparse
+from argparse import RawTextHelpFormatter
+from Bio.Seq import Seq
+from Bio.Alphabet import generic_dna
+import pysam
+
+description="Reverses the allele for variants that got mapped onto the reverse strand of the new genome, and prints \
+everything to a new file\n"
+
+parser = argparse.ArgumentParser(description = description, formatter_class=RawTextHelpFormatter)
+parser.add_argument("-i", "--bam", help="bam file containing remapped variants ")
+parser.add_argument("-o", "--outfile", help="name of new file")
+parser.add_argument("-p", "--old_ref_alleles", help="name of output old ref alleles")
+args = parser.parse_args()
+
+
+bamfile=pysam.AlignmentFile(args.bam, 'rb')
+outfile=open(args.outfile, 'w')
+old_ref_alleles=open(args.old_ref_alleles, 'w')
+
+# Reverses the allele for variants that got mapped onto the reverse strand of the new genome, and prints everything 
+# into correct columns
+
+for read in bamfile:
+	name = read.query_name
+	info = name.split("|")
+	nucl = info[3]
+	# Mapped onto reverse strand:
+	if read.flag & 16: # Decoding bitwise flag with &; may need to be changed back to ==
+		nucl = Seq(nucl, generic_dna).complement()
+	# Write it all to the file:
+	if not(read.flag & 4): # 4 means unmapped
```

Comment:
These can be checked more directly using [`read.is_reverse`](https://pysam.readthedocs.io/en/latest/api.html#pysam.AlignedSegment.is_reverse) and [`read.is_unmapped`](https://pysam.readthedocs.io/en/latest/api.html#pysam.AlignedSegment.is_unmapped) attributes